sshconnect.c

```diff
@@ -513,7 +519,7 @@ ssh_connect_direct(struct ssh *ssh, const char *host, struct addrinfo *aitop,
 
 	/* Return failure if we didn't get a successful connection. */
 	if (sock == -1) {
-		error("ssh: connect to host %s port %s: %s",
+		error("ssh: connect to %s%s: %s",
 		    host, strport, errno == 0 ? "failure" : strerror(errno));
```

Comment:
Adapt error message for unix paths.

Reply:
It's not needed here. For a UNIX socket, `strport` will be an empty string, so `%s%s` will just print what's in `host`, which contains the path. See:
```
% /usr/local/bin/ssh okurth@/foo.sock
ssh: connect to /foo.sock: No such file or directory
```

Reply:
Yes, but it is broken for IP addresses:
```
$ ./ssh user@localhost -p 3333
ssh: connect to localhost3333: Connection refused
```

Reply:
Oh, I missed that, thanks! This is from one of pjd's original changes.

And fixed, with 5e813709

Reply:
I think pjd's original changes were fine, but later destroyed by my attempts to simplify it :/ Thanks for restoring!